projects/lesson-5/runtime/src/kitties.rs

```diff
@@ -0,0 +1,374 @@
+			<OwnedKitties<T>>::append(&to,kitty_id);
+			<OwnedKitties<T>>::remove(&sender,kitty_id);
+		}
+	}
+}
+
+impl<T: Trait> OwnedKitties<T> {
+    fn read_head(account: &T::AccountId) -> KittyLinkedItem<T> {
+        Self::read(account, None)
+    }
+
+    fn write_head(account: &T::AccountId, item: KittyLinkedItem<T>) {
+        Self::write(account, None, item);
+    }
+
+    fn read(account: &T::AccountId, key: Option<T::KittyIndex>) -> KittyLinkedItem<T> {
+        <OwnedKitties<T>>::get(&(account.clone(), key)).unwrap_or_else(|| KittyLinkedItem {
+            prev: None,
+            next: None,
+        })
+    }
+
+    fn write(account: &T::AccountId, key: Option<T::KittyIndex>, item: KittyLinkedItem<T>) {
+        <OwnedKitties<T>>::insert(&(account.clone(), key), item);
+    }
+
+    pub fn append(account: &T::AccountId, kitty_id: T::KittyIndex) {
+        let head = Self::read_head(account);
+        let new_head = KittyLinkedItem {
+            prev: Some(kitty_id),
+            next: head.next,
+        };
+
+        Self::write_head(account, new_head);
+
+        let prev = Self::read(account, head.prev);
+        let new_prev = KittyLinkedItem {
+            prev: prev.prev,
+            next: Some(kitty_id),
+        };
+        Self::write(account, head.prev, new_prev);
+
+        let item = KittyLinkedItem {
+            prev: head.prev,
+            next: None,
+        };
+        Self::write(account, Some(kitty_id), item);
+    }
+
+    pub fn remove(account: &T::AccountId, kitty_id: T::KittyIndex) {
+        if let Some(item) = <OwnedKitties<T>>::take(&(account.clone(), Some(kitty_id))) {
+            let prev = Self::read(account, item.prev);
+            let new_prev = KittyLinkedItem {
+                prev: prev.prev,
+                next: item.next,
+            };
+
+            Self::write(account, item.prev, new_prev);
+
+            let next = Self::read(account, item.next);
+            let new_next = KittyLinkedItem {
+                prev: item.prev,
+                next: next.next,
+            };
+
+            Self::write(account, item.next, new_next);
+        }
+    }
+}
+
+fn combine_dna(dna1: u8, dna2: u8, selector: u8) -> u8 {
+    ((selector & dna1) | (!selector & dna2))
+}
+
+impl<T: Trait> Module<T> {
+    fn random_value(sender: &T::AccountId) -> [u8; 16] {
+        let payload = (<system::Module<T>>::random_seed(), sender, <system::Module<T>>::extrinsic_index(), <system::Module<T>>::block_number());
+        payload.using_encoded(blake2_128)
+    }
+
+    fn next_kitty_id() -> result::Result<T::KittyIndex, &'static str> {
+        let kitty_id = Self::kitties_count();
+        if kitty_id == T::KittyIndex::max_value() {
+            return Err("Kitties count overflow");
+        }
+        Ok(kitty_id)
+    }
+
+    fn insert_owned_kitty(owner: &T::AccountId, kitty_id: T::KittyIndex) {
+        // 作业：调用 OwnedKitties::append 完成实现
+        <OwnedKitties<T>>::append(owner,kitty_id);
```

Comment:
You need to update kitty owner:
`<KittyOwner<T>>::insert(kitty_id, owner.clone());`